src/components/canrawsender/newlinemanager.h

```diff
@@ -0,0 +1,132 @@
+#ifndef NEWLINEMANAGER_H
+#define NEWLINEMANAGER_H
+
+#include <QCheckBox>
+#include <QHBoxLayout>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QTimer>
+#include <QValidator>
+#include <QtSerialBus/QCanBusFrame>
+
+class CanRawSender;
+/// \class Iterator
+/// \brief This class provide implementation of iterator used to correctly work for loop with enum class
+/// \tparam typeName Name of enum class
+/// \tparam beginVal Begin value of enum
+/// \tparam endVal End value of enum
+template <typename typeName, typeName beginVal, typeName endVal> class Iterator {
```

Comment:
Some types must be exposed to make this iterator STL-compliant, see member types here http://en.cppreference.com/w/cpp/iterator/iterator_traits